packages/server/src/threads/automation.ts

```diff
@@ -385,7 +385,7 @@ class Orchestrator {
     stepIdx: number,
     pathIdx?: number
   ): Promise<number> {
-    await processObject(loopStep.inputs, this.processContext(this.context))
+    await processObject(loopStep.inputs, this.mergeContexts(this.context))
```

Comment:
What is this change doing? Is there anyway we can test for why this change was required?

Reply:
So two changes 

- First I renamed processContext -> mergeContext as it makes much sense.
- The actual fix is on line 648, stepsById / steps / stepsByName contexts were not being merged together before being sent to the step function. This meant that specifically for JS Scripting block, that when it tried to use it's own `processObject` the context which it was looking for was not there. I can get a test in for it